src/addrman.h

```diff
@@ -332,7 +339,12 @@ friend class CAddrManTest;
         OverrideStream<Stream> s(&s_, s_.GetType(), s_.GetVersion() | ADDRV2_FORMAT);
 
         s << static_cast<uint8_t>(Format::V3_BIP155);
```

Comment:
unrelated style nit: Any reason why this can't use `CustomUintFormatter` (and its run-time sanity checks)?

Reply:
It is possible to use `CustomUintFormatter` here. This line is not otherwise modified by the patch, so it would be sneaking an unnecessary change, so I left it as is to ease reviewers and future history observers.

Reply:
_Now_ I changed this line for other reasons and switched to `CustomUintFormatter` :-)

Reply:
re https://github.com/bitcoin/bitcoin/pull/20284#discussion_r520327284. I don't understand why we want this change. The comment for `CustomUintFormatter` says "This is only intended to implement serializers that are compatible with existing formats, and its use is not recommended for new data structures."

Reply:
The comment only applies to endianness, see https://github.com/bitcoin/bitcoin/pull/18317#discussion_r427621184.
Generally, I think serialize/deserialze should use the same helpers, so either `CustomUintFormatter` should be used for both paths or for none.